src/schema_compile_node.c

```diff
@@ -4223,6 +4223,7 @@ lys_compile_grouping(struct lysc_ctx *ctx, struct lysp_node *pnode, struct lysp_
     len = lys_compile_grouping_pathlog(ctx, grp->parent, &path);
     if (len < 0) {
         LOGMEM(ctx->ctx);
+        free(path);
```

Comment:
I have tried to fix this properly with the last `devel` commit so it should not be needed anymore.